improve the AWS codelab (#365)
transparency-dev/trillian-tessera: deployment/live/aws/codelab/README.md

# AWS codelab deployment

> This codelab helps you bring a test Trillian Tessera infrastructure on AWS, and

Reviewer: ```suggestion
This codelab helps you bring up a test Trillian Tessera infrastructure on AWS, and
```

> to use it running a test personality server on an EC2 VM. The infrastructure 

Reviewer: ```suggestion
to use it by running a test personality server on an EC2 VM. The infrastructure 
```

    ```

 1. Deploy the infrastructure:
    ```
    terragrunt apply --terragrunt-working-dir=deployment/live/aws/codelab/
    ```
    This brings up the Terraform infrastructure (S3 bucket + DynamoDB table for
    terraform state locking only) and the Trillian Tessera infrastructure: an
    RDS Aurora instance, a private S3 bucket, and connects this bucket to the
    default VPC that your VM should be connected to.

 1. Save the RDS instance URI and S3 bucket name for later:
    ```
    export LOG_RDS_DB=$(terragrunt output --terragrunt-working-dir=deployment/live/aws/codelab/ --raw log_rds_db)
    export LOG_BUCKET=$(terragrunt output --terragrunt-working-dir=deployment/live/aws/codelab/ --raw log_bucket_id)
    export LOG_NAME=$(terragrunt output --terragrunt-working-dir=deployment/live/aws/codelab/ --raw log_name)
    ```
 
1. Connect the VM and Aurora database following [these instructions](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/tutorial-ec2-rds-option1.html#option1-task3-connect-ec2-instance-to-rds-database),
   it takes a few clicks in the UI.

### Start a Trillian Tessera personality
A personality is a server that interacts with Trillian Tessera's storage

> infrastructure. In this codelab, it accepts writes requests on a `add/` HTTP

Reviewer: /write/POST/

              --out_priv=/home/ec2-user/tessera-keys/$LOG_NAME.sec \
              --out_pub=/home/ec2-user/tessera-keys/$LOG_NAME.pub
    ```

>  1. Generate and copy these environment variale definitions. \

Reviewer: ```suggestion
 1. Generate and copy these environment varibale definitions. \
```

    They will allow you
 to send requests to the log from a different terminal once it's running:

>     ```
>     echo -e "\n\n"
>     echo =================================================================================================================
>     echo "Copy these variable definitions to use in a different terminal:"
>     echo -e "\n"
>     echo "export WRITE_URL=http://localhost:2024/"
>     echo "export READ_URL=https://$LOG_BUCKET.s3.$AWS_REGION.amazonaws.com/"
>     echo "export LOG_PUBLIC_KEY=$(cat /home/ec2-user/tessera-keys/$LOG_NAME.pub)"
>     echo =================================================================================================================
>     echo -e "\n\n"
>     ```

Reviewer: I found this a bit confusing, but I understand what you're doing now. How about tweaking the wording a bit & removing unnecessary fluff in the `bash`?

Also, consider moving this section below the "Run the conformance binary" step if possible - I think that'll help with mental "flow" through these steps.

> Running the commands below will print some easily copy-and-pasteable `export`s which you can use to set up the environment in a second terminal ready to be able to send requests:

```suggestion
    echo "export WRITE_URL=http://localhost:2024/"
    echo "export READ_URL=https://$LOG_BUCKET.s3.$AWS_REGION.amazonaws.com/"
    echo "export LOG_PUBLIC_KEY=$(cat /home/ec2-user/tessera-keys/$LOG_NAME.pub)"
```

Author: I like your wording, thanks. It can't be moved below because Tessera will be running, so folks would have to kill Tessera in order to run these commands. Alternatievly, we could kick Tessera in the background but, that feels worse.

 1. Run the Conformance personality binary.
    ```
    go run ./cmd/conformance/aws \
      --bucket=$LOG_BUCKET \
      --db_user=root \
      --db_password=password \
      --db_name=tessera \
      --db_host=$LOG_RDS_DB \
      --signer=$(cat /home/ec2-user/tessera-keys/$LOG_NAME.sec)
      -v=3
    ```

 1. 🎉 **Congratulations** 🎉

>     You have successfully brought up a Trillian Tessera

Reviewer: ```suggestion
    You have successfully brought up Trillian Tessera
```

    infrastructure on AWS, and started a personality server that can add entries to it.

    Use the environment variables from above to interact with the personality in a different terminal.

>     This personality accepts `add/` requests at `WRITE_URL`.

Reviewer: ```suggestion
    This personality accepts `POST` requests to the `/add` endpoint under `WRITE_URL`.
```
